pkg/pgmodel/querier/series_set.go

```diff
@@ -25,21 +24,38 @@ const (
 
 // pgxSeriesSet implements storage.SeriesSet.
 type pgxSeriesSet struct {
-	rowIdx  int
-	rows    []timescaleRow
-	err     error
-	querier labelQuerier
+	rowIdx     int
+	rows       []timescaleRow
+	labelIDMap map[int64]*labels.Label
```

Comment:
Is it necessary for this to be a pointer? If not, we can save some allocation here even if this is not expensive.